internal/controllers/machineset/machineset_controller.go

```diff
@@ -1059,31 +1158,38 @@ func (r *Reconciler) getMachineNode(ctx context.Context, cluster *clusterv1.Clus
 	return node, nil
 }
 
-func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *clusterv1.Cluster, ms *clusterv1.MachineSet, filteredMachines []*clusterv1.Machine) (ctrl.Result, error) {
+func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, s *scope) (ctrl.Result, error) {
+	if s.machines == nil {
+		return ctrl.Result{}, nil
+	}
+
 	log := ctrl.LoggerFrom(ctx)
 
 	// Calculate how many in flight machines we should remediate.
 	// By default, we allow all machines to be remediated at the same time.
-	maxInFlight := len(filteredMachines)
+	maxInFlight := len(s.machines)
 
 	// If the MachineSet is part of a MachineDeployment, only allow remediations if
 	// it's the desired revision.
-	if r.isDeploymentChild(ms) {
-		owner, err := r.getOwnerMachineDeployment(ctx, ms)
-		if err != nil {
-			return ctrl.Result{}, err
+	if isDeploymentChild(s.machineSet) {
+		if s.owningMachineDeployment == nil {
+			var err error
+			s.owningMachineDeployment, err = r.getOwnerMachineDeployment(ctx, s.machineSet)
```

Comment:
What is the reason that we do this here again? (it was already done in the reconcile func)

Reply:
jep, forgot to remove this one :-) should already be set.